packages/dev/core/src/Meshes/Compression/dracoDecoder.ts

```diff
@@ -0,0 +1,282 @@
+
+interface MeshData {
+    indices?: Uint16Array | Uint32Array;
+    attributes: Array<AttributeData>;
+    totalVertices: number;
+}
+
+/**
+ * Draco compression (https://google.github.io/draco/)
+ *
+ * This class wraps the Draco decoder module.
+ *
+ * By default, the configuration points to a copy of the Draco decoder files for glTF from the Babylon.js preview cdn https://preview.babylonjs.com/draco_wasm_wrapper_gltf.js.
+ *
+ * To update the configuration, use the following code:
+ * ```javascript
+ *     DracoDecoder.Config = {
+ *          wasmUrl: "<url to the WebAssembly library>",
+ *          wasmBinaryUrl: "<url to the WebAssembly binary>",
+ *          fallbackUrl: "<url to the fallback JavaScript library>",
+ *     };
+ * ```
+ *
+ * Draco has two versions, one for WebAssembly and one for JavaScript. The decoder configuration can be set to only support WebAssembly or only support the JavaScript version.
+ * Decoding will automatically fallback to the JavaScript version if WebAssembly version is not configured or if WebAssembly is not supported by the browser.
+ * Use `DracoDecoder.Available` to determine if the decoder configuration is available for the current context.
+ *
+ * To decode Draco compressed data, get the default DracoDecoder object and call decodeMeshToGeometryAsync:
+ * ```javascript
+ *     var geometry = await DracoDecoder.Default.decodeMeshToGeometryAsync(data);
+ * ```
+ */
+export class DracoDecoder extends DracoCodec<DecoderModule> {
+    protected static override _Default: Nullable<DracoDecoder> = null;
+    /**
+     * Default instance for the DracoDecoder.
+     */
+    public static get Default(): DracoDecoder {
+        if (!DracoDecoder._Default) {
+            DracoDecoder._Default = new DracoDecoder();
+        }
+        return DracoDecoder._Default;
+    }
+
+    /**
+     * Configuration for the DracoDecoder. Defaults to the following:
+     * - numWorkers: 50% of the available logical processors, capped to 4. If no logical processors are available, defaults to 1.
+     * - wasmUrl: `"https://cdn.babylonjs.com/draco_wasm_wrapper_gltf.js"`
+     * - wasmBinaryUrl: `"https://cdn.babylonjs.com/draco_decoder_gltf.wasm"`
+     * - fallbackUrl: `"https://cdn.babylonjs.com/draco_decoder_gltf.js"`
+     */
+    public static override Config: IDracoCodecConfiguration = {
+        wasmUrl: `${Tools._DefaultCdnUrl}/draco_wasm_wrapper_gltf.js`,
+        wasmBinaryUrl: `${Tools._DefaultCdnUrl}/draco_decoder_gltf.wasm`,
+        fallbackUrl: `${Tools._DefaultCdnUrl}/draco_decoder_gltf.js`,
+    };
+
+    protected override _isModuleAvailable(): boolean {
+        return typeof DracoDecoderModule !== "undefined";
+    }
+
+    protected override _createModuleAsync(wasmBinary?: ArrayBuffer, jsModule?: any): Promise<{ module: DecoderModule }> {
+        return new Promise((resolve) => {
+            ((jsModule as DracoDecoderModule) || DracoDecoderModule)({ wasmBinary }).then((module) => {

```

Comment:
Maybe this is just pre-existing refactored code, but it would be nice to replace all of these `.then` with `await`.